### DISCUSSION TITLE
VSCODE view()->layout() inside render gives undefined method.

### DISCUSSION THREAD

**Discussion by fenilli** [livewire/livewire#2089] [1. Help] [answered]
Inteliphense doesn't know there is a layout inside of view because of return types, is there a way to make it explicity so the ide knows where to look?

**Comment by joshhanley** [comment, marked as answer]
Your best option would be to lodge an issue on the inteliphense repo.

The reason it's not detected is because the layout method has been applied to the view using a macro as its a custom method. So the inteliphense plugin would need to know about and support macros.

Hope this helps!

**Reply by DickiGendraErlangga** [reply]
what support macros can know about intelphense plugin ?

**Comment by joshhanley** [comment]
Your best option would be to lodge an issue on the inteliphense repo.

The reason it's not detected is because the layout method has been applied to the view using a macro as its a custom method. So the inteliphense plugin would need to know about and support macros.

Hope this helps!
